### ISSUE TITLE
"load" after "download" is failing when triggered by cron

### ISSUE THREAD

**Issue by ratsbane** [ratsbane/airwho-loader#4]
copy_and_unzip_file_into_temp: Loading /var/aircraft/2019/2019-03-13_07_15_19 into temp as 2019-03-13.zip
    unzip /var/aircraft/temp/2019-03-13.zip -d /var/aircraft/temp
  End-of-central-directory signature not found.  Either this file is not .
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
    unzip:  cannot find zipfile directory in one of /var/aircraft/temp/2019-03-13.zip or
        /var/aircraft/temp/2019-03-13.zip.zip, and cannot find /var/aircraft/temp/2019-03-13.zip.ZIP, period.

**Comment by ratsbane** [comment]
- When I manually copy the file over from archive path to temp and then unzip, there are no problems
- The zip file copied to the temp directory by the cron - download - load sequence is shorter than the file in the archive directory.
- When I compare the file copied to temp with the archive file, they are the same up to the eof of the file in temp:
    doug@airwho:/var/aircraft/bin$ cmp -l /var/aircraft/2019/2019-03-13_07_15_19 /var/aircraft/temp/2019-03-13.zip | gawk '{printf "%08X %02X %02X\n", $1, strtonum(0$2), strtonum(0$3)}'
    cmp: EOF on /var/aircraft/temp/2019-03-13.zip after byte 64790528

This leads to the theory that the download process, which writes the file to disk, is not through writing by the time that the system call to "load" happens.  "download" is a Python program.  It uses file.close() after writing, which Python documentation indicates does call flush() and fsync() https://docs.python.org/3/library/io.html#io.IOBase.close  I did consider that using a Python "with" context block might be nice, but there's nothing _wrong_ with explicitly calling f.close()
- Alternate theory - cron is starting with a path other than what I'm using to test - it's copying/unzipping a file in a different location.  The timestamp on the zip file in temp seems to disprove this theory - although the _name_ of the zip file in temp supports it (e.g., the file downloaded was /2019/2019-03-13_08_11_41 but the zip file name, after I ran it manually:

-rw-rw-r--  1 doug doug  64751386 Mar 13 08:12 2019-03-09.zip

In any case, the load script needs to bail if the zip is unsuccessful and the download script should accept a return code from that and, if it's not ok, stop before calling load_history
